js/ui/file_manager/ui.file_manager.common.js

```diff
@@ -22,34 +30,31 @@ const whenSome = function(arg, onSuccess, onError) {
     };
 
     if(!Array.isArray(arg)) {
-        return when(arg)
-            .then(onSuccess,
-                error => {
-                    if(error) {
-                        onError(createErrorInfo(0, error));
-                    }
-                });
+        arg = [ arg ];
     }
 
     const deferreds = arg.map((item, index) => {
         return when(item)
-            .then(result => createResult(result, true),
+            .then(
+                result => {
+                    const resultObj = createResult(result, true);
+                    if(resultObj.error && onError) {
+                        onError(createErrorInfo(index, resultObj.error));
+                    }
+
+                    if(resultObj.success && onSuccess) {
+                        onSuccess();
+                    }
+                },
                 error => {
                     if(error) {
                         onError(createErrorInfo(index, error));
                     }
-                    return createResult(null, false, !error, error);
                 });
     });
```

Comment:
I found that the new implementation does not return `Deffered ` as a result. Without this, it is impossible to achieve the desired in [this comment](https://github.com/DevExpress/DevExtreme/pull/8365#discussion_r293697464).

Reply:
Please, review new version of code.

Reply:
It's good.